server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java

```diff
@@ -218,4 +217,22 @@ public OshiSysMonitor getOshiSysMonitor(
       return new OshiSysMonitor(dimensions, oshiSysConfig);
     }
   }
+
+  /**
+   * Checks if a monitor needs to be loaded on this service based on its node role.
+   */
+  private boolean shouldLoadMonitor(Class<?> monitorClass, Set<NodeRole> nodeRoles)
+  {
+    final LoadScope loadScope = monitorClass.getAnnotation(LoadScope.class);
```

Comment:
Any desire to get fancy here and check superclass annotations too? Probably not necessary but just something that came to mind.

Reply:
Sure, that can be done.

I guess we would need to check the superclass recursively if there is no annotation defined at the current class level. Would that make sense?

Reply:
Thanks! Updated to handle annotations on super-class and added tests.